packages/opentelemetry-exporter-collector-grpc/src/util.ts

```diff
@@ -108,8 +108,12 @@ export function send<ExportItem, ServiceRequest>(
 }
 
 export function validateAndNormalizeUrl(url: string): string {
+  const hasProtocol = url.match(/^([\w]{1,8}):\/\//);
+  if (!hasProtocol) {
+    url = `https://${url}`;
```

Comment:
Shouldn't this be grpc not https?

Reply:
funny story if you do grpc, then browser js tests fail because browser URL library (unlike nodejs library) has no idea how to deal with "grpc://"

Reply:
seriously go to chromium dev tools and try `let url = URL("grpc://somehost:port/path");` and see what it does

Reply:
`let url = new URL("grpc://somehost:port/path");` works fine in chrome 91.0.4472.114

![image](https://user-images.githubusercontent.com/1612643/125333594-6baaad80-e318-11eb-89fc-3e8565e81857.png)

Reply:
I wonder what version is the "bad" version. In any case it's fine if the browser tests fail, but it's frustrating to me since grpc is the "correct" protocol.

Reply:
![Screenshot from 2021-07-12 10-56-59](https://user-images.githubusercontent.com/614704/125334227-fa5f0080-e2ff-11eb-8bc1-55c88bcc03a7.png)

it becomes unable to parse the hostname from the path.

Reply:
aha thats what i was missing. still frustrating but i guess this is a fine workaround since the protocol doesn't actually change the behavior at all. Then what happens if the user explicitly sets `grpc://`? Do we break?

Reply:
correct, we do break in the browser. (and this was part of my original pushback against using the URL library in the first place in #2130, heh, if we're having to work around the limitations of the URL library)

Reply:
Hmm I must not have fully understood that at the time we introduced #2130 

For now, this is a decent workaround but we should fix this in the future.

Reply:
but why the grpc collector which is meant only for node not for browser needs to deal with browser tests ?

Reply:
It's not, I just checked. The browser tests don't run in the grpc exporter.

Reply:
@lizthegrey why do you test that in browser then, as it is not meant to work there anyway ?

Reply:
I thought that it _was_ intended to be possible for frontend instrumentation to be able to write data to a grpc collector?

I was testing using a browser js shell to verify my understanding of the URL class, and found this divergence between nodejs and browser behavior :(

in any event the protocol is dropped when constructing the GRPC client and only hostname:port is passed anyways so... it doesn't matter?

Reply:
grpc is not supported in the browser currently. Afaik only json over http is supported but @obecny is probably a better source of truth in that arena.

Reply:
In browser you can only use json version of collector exporter , proto and grpc are for node.js only. Not sure then how would browser send data to grpc collector - it can only send to json

Reply:
Currently yes but the proto exporter could be fixed to works in browser i believe since it uses `protobufjs` which works on browser

Reply:
Is resolving this blocking or should we proceed with merging?

Reply:
@lizthegrey I believe we could update the `grpc://` protocol here, but you mentionned that tests were failing in this case ? the grpc collector doesn't run tests in browser so that seems weird

Reply:
maybe the browser tests failed just accidentally then, is this repeatable failure ???

Reply:
no, the browser tests don't run on this module so I'm saying that if they _were_ run it would fail (since I manually ran the test code in chromium dev tools), but there's nothing stopping me from making the change and having CI pass, I just worry about deliberately introducing something we know will break (for mysterious reasons) in a browser

Reply:
I don't think its blocking then, this package use grpc which would never be able to be built for a web target anyway

Reply:
It's not blocking since it isn't used anyway